false positive with `Undefined name Pool F821` in .pyi

minimal reproducible example of [a larger module type stub](https://github.com/chdsbd/kodiak/blob/0c1171bc8f574d7ee11ab6174036edadfbc5ec09/bot/typings/asyncio_redis/pool.pyi#L4-L4) aka `.pyi` file:


```python
# pool.pyi
class Pool:
    @classmethod
    def create(cls) -> Pool: ...
```

errors with:

```
Undefined name `Pool`Ruff(F821)
```

```
typings/asyncio_redis/pool.pyi:31:10: F821 Undefined name `Pool`
```

--- a/resources/test/fixtures/pyflakes/F821_8.pyi
+++ b/resources/test/fixtures/pyflakes/F821_8.pyi
@@ -1,0 +1,3 @@
+class Pool:
+    @classmethod
+    def create(cls) -> Pool: ...

--- a/src/checkers/ast.rs
+++ b/src/checkers/ast.rs
@@ -151,7 +151,7 @@ impl<'a> Checker<'a> {
             in_subscript: false,
             seen_import_boundary: false,
             futures_allowed: true,
-            annotations_future_enabled: false,
+            annotations_future_enabled: path.extension().map_or(false, |ext| ext == "pyi"),
             except_handlers: vec![],
             // Check-specific state.
             flake8_bugbear_seen: vec![],

--- a/src/pyflakes/mod.rs
+++ b/src/pyflakes/mod.rs
@@ -98,6 +98,7 @@ mod tests {
     #[test_case(CheckCode::F821, Path::new("F821_5.py"); "F821_5")]
     #[test_case(CheckCode::F821, Path::new("F821_6.py"); "F821_6")]
     #[test_case(CheckCode::F821, Path::new("F821_7.py"); "F821_7")]
+    #[test_case(CheckCode::F821, Path::new("F821_8.pyi"); "F821_8")]
     #[test_case(CheckCode::F822, Path::new("F822.py"); "F822")]
     #[test_case(CheckCode::F823, Path::new("F823.py"); "F823")]
     #[test_case(CheckCode::F841, Path::new("F841_0.py"); "F841_0")]

